vignettes/dsl-errors.Rmd

```diff
@@ -0,0 +1,123 @@
+---
+title: "DSL parse errors"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{DSL parse errors}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+<!-- See comments at the top of odin's error vignette for details -->
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette outlines errors that might be generated when parsing mcstate2 DSL code, with more explanation about the error and how they can be avoided.  Don't read this top to bottom as it's quite boring!  However, if we get errors that benefit from more explanation about why they've been thrown then we'll expand on the contents here and arrange for these to be linked from the thrown error directly.
+
+The error numbers are arbitrary after the first digit.  The first digit will correspond to different phases of the parsing:
+
+* `E1xx` - errors during parsing of individual expressions
+* `E2xx` - errors when considering the system as a whole
+
+# `E101`
+
+We found an expression that is neither an assignment (with `<-`) or a stochastic relationship (with `~`)
+
+Example:
+
+```
+a + 1
+```
+
+# `E102`
+
+Invalid left hand side of a relationship operator (`~`).  Currently the left hand side must be symbol, though this will be relaxed in future once we support an array syntax.  However, you may not use things like numbers or function calls on the left hand side.
```

Comment:
```suggestion
Invalid left hand side of a relationship operator (`~`).  Currently the left hand side must be symbol, though this will be relaxed in the future once we support an array syntax.  However, you may not use things like numbers or function calls on the left hand side.
```